src/ActionMailerNext.Standalone/Helpers/HandlebarsFilesTemplateResolver.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace ActionMailerNext.Standalone.Helpers
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class HandlebarsFilesTemplateResolver : ITemplateResolver
+    {
+        private readonly string _viewPath;
+
+        /// <summary>
+        ///     Creates a template resolver using the specified path.  If no path is given, this defaults to "Views"
+        /// </summary>
+        /// <param name="viewPath">The path containing your views</param>
+        public HandlebarsFilesTemplateResolver(string viewPath)
+        {
+            _viewPath = viewPath;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        public string Resolve(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+                return null;
+
+            string csViewName = name;
+
+            if (!csViewName.EndsWith(".hbs"))
+                csViewName += ".hbs";
+
+            var appRoot = AppDomain.CurrentDomain.BaseDirectory;
+
+            var csViewPath = (csViewName).StartsWith("~")
+                ? Path.GetFullPath(Path.Combine(appRoot, csViewName.Substring(2)))
+                : Path.GetFullPath(Path.Combine(appRoot, _viewPath, csViewName));
+
+            //Works with forward and backward slashes in the path
+            if (File.Exists(csViewPath))
+            {
+                return File.ReadAllText(csViewPath);
+            }
+            throw new TemplateResolvingException { SearchPaths = new List<string> { csViewPath } };
+        }
+
+        public List<MailTemplate> GetAllPartialTemplates()
+        {
+            return GetAllTemplates().Where(t => t.IsPartial).ToList();
+        }
+
+        private IEnumerable<MailTemplate> GetAllTemplates()
+        {
+            var templates = SearchTemplates("*");
+            return templates;
+        }
+
+        private List<MailTemplate> SearchTemplates(string searchPattern)
+        {
+            searchPattern = $"*{searchPattern}*.hbs";
+            var templatesDir = Path.Combine(Directory.GetParent(AppDomain.CurrentDomain.BaseDirectory).FullName, _viewPath);
```

Comment:
I am not sure about this search pattern but sometimes you have a template e.g. Result.html and ResultMan.html

Reply:
i renamed all the files to *.hbs ... 
about the view names that we sometimes send it as ...html but at another point we remove the extensions before searching for the file